File: backend/src/main/java/com/org/playboard/dto/stats/PlayerAttendanceDto.java
package com.org.playboard.dto.stats;

import java.time.Instant;
import java.util.List;

/**
 * Backs the Profile attendance calendar: the distinct match instants a player was in
 * within the queried window. Instants serialize to ISO-8601 strings; the client buckets
 * them into local calendar days to paint the month grid.
 */
public record PlayerAttendanceDto(List<Instant> playedAt) {}

File: android/app/src/main/java/com/org/playboard/data/remote/dto/PlayerAttendanceDto.kt
package com.org.playboard.data.remote.dto

import kotlinx.serialization.Serializable

/**
 * `GET /groups/{id}/members/{userId}/attendance` — the distinct match instants (ISO-8601)
 * a player was in within the queried window. The client buckets these into local calendar
 * days to paint the Profile attendance grid.
 */
@Serializable
data class PlayerAttendanceDto(
    val playedAt: List<String> = emptyList(),
)
